### PR TITLE
ES|QL: Add 'preview' information to functions docs for Kibana

### PR DESCRIPTION
**Pull request by luigidellaquila**
Sending Kibana information to know if a function is still in preview mode (ie. not production ready)

Related to #112755

### CODE REVIEW

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java
Comment (ivancea) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java:1]: I see there's only 1 function (`VALUES`) using preview. I suppose it should be reviewed, maybe for another PR

**Review by ivancea — APPROVED** [review]
Nice!

**Comment by stratoula** [comment]
Nice!

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java
Comment (luigidellaquila) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java:1]: Yes, for now it's only VALUES, but I see there is an hardcoded `"Do not use VALUES on production environments."`, let me fix it

Comment (ivancea) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java:1]: Uh... Don't blame the guy who added that, he was young and inexperienced

**Review by ivancea — APPROVED** [review]

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java [hunk 1]
```diff
@@ -1148,6 +1148,7 @@ private static void renderKibanaFunctionDefinition(
             }
             builder.endArray();
         }
+        builder.field("preview", info.preview());
```
Comment (bpintea) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java:1151]: Nit: wondering if we could/should add `preview` only then when it's `true`. Though "explicit is better than implicit".

Comment (luigidellaquila) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java:1151]: I'll let Kibana folks decide here, cc @drewdaemon

**Review by nik9000 — APPROVED** [review]

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java (hunk 1 above)
Comment (drewdaemon) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java:1151]: No preference here.

**Review by drewdaemon — APPROVED** [review]
Looks great. Any function you mark as `preview` will not be suggested in the ES|QL editor in any context (snapshot builds or no).

We could consider being smarter about it, but I'd need guidance from y'all on how to reliably detect when "preview" functions should be shown.

**Pull request merged**